### PR TITLE
514 track events

### LINKED ISSUE #514: Add new tracking events
As a marketer
So that I can more closely track how people are interacting with pages
I want to a js library to can send custom analytics events to segment based on when the user clicks on parts of the page

At the moment this is done as a [one-off in all.js](https://github.com/makersacademy/website/blob/master/source/javascripts/all.js#L40) but we need this more generically as it's required a lot on the ronin website.

Once it's done we should track the following events on the website:

- [ ] Grad slider clicked
- [ ] Curriculum slider clicked
- [ ] Alumni videos watched

### PR DESCRIPTION
**Pull request by jafrog**
Close #514 

Now to track new event either add selector and event type to [`trackEvents` object](https://github.com/makersacademy/website/blob/bfce789dc9b735d8008671c04dbef36c989ccf56/source/javascripts/all.js#L40-L46) or add `data-track = <EVENT TYPE>` to an element

### CODE REVIEW

File: source/javascripts/all.js [hunk 1]
```diff
@@ -37,5 +37,33 @@ $(document).ready(function() {
...
+    ".graduates-video": "Watched Alumni Video",
+    "[data-track]": null,
+  }
+
+  function isAnchor(el) {
+    el.nodeName == "A";
+  }
+
+  for (var el in trackEvents) {
+    var properties = {"Course Type": "Offline"};
```
Comment (danldb) [inline source/javascripts/all.js]: Small suggestions:

Could declare this var outside the loop so it is not declared in every iteration.

`properties` is not very descriptive - maybe `courseProperties`?

**Comment by danldb** [comment]
Build failing at the moment

**Comment by silvabox** [comment]
We bumped the Ruby version to 2.2.3 but Codeship was still using 2.2.2.  It's passing now.

**Comment by danldb** [comment]
:ship:

**Pull request merged**